File: fsharp-fuel-engine/FuelUpload.Domain/Vehicle.fs
namespace FuelUpload.Domain

[<CLIMutable>]
type Vehicle =
    { VehicleId: string
      Registration: string }

[<RequireQualifiedAccess>]
type VehicleLookupResult =
    | Matched of Vehicle
    | NotFound
    | Ambiguous of Vehicle list
    | Fatal of FatalProcessingError

File: haskell-fuel-engine/src/FuelUpload/Domain/Vehicle.hs
module FuelUpload.Domain.Vehicle
  ( Vehicle (..)
  , VehicleLookupResult (..)
  ) where

import FuelUpload.Domain.Primitive

data Vehicle = Vehicle
  { vehicleId :: VehicleId
  , vehicleRegistration :: Registration
  }
  deriving stock (Eq, Show)

data VehicleLookupResult
  = VehicleFound Vehicle
  | VehicleMissing Registration
  | VehicleLookupFatal FatalError
  deriving stock (Eq, Show)
